packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -58,6 +59,14 @@ export type BaseInputProps = FormInputLabelProps &
      * The callback function to be invoked when the value of the input field changes
      */
     onChange?: FormInputOnEvent;
+    /**
+     * The callback function to be invoked when the value of the input field has any input
+     */
+    onInput?: FormInputOnEvent;
```

Comment:
Why do we need `onInput` along with `onChange` here 🤔 
Historically I think in React `onChange` works like `onInput`

Reply:
https://github.com/razorpay/blade/pull/731#discussion_r986712992

Because of this  ^

Reply:
Oh nice, missed this in diff view. We can perhaps add a comment in the code as well